### ISSUE TITLE
WF Objs have UV-mappings parsed incorrectly

### ISSUE THREAD

**Issue by Hibiii** [leezer3/OpenBVE#740]
### Description

Only recently I realized I could use [Blockbench](https://blockbench.net/) to make models and export in Wavefront format, which would greatly help me make models (sincerely tired from typing `AddVertex`s).

ObjectViewer doesn't parse UV mappings correctly from `obj`s. Any textured `obj` has its UV mapping messed up. The mesh is loaded correctly though. Also, it logs a warning as it doesn't recognize `o`. This is not an issue with Blockbench as `obj`s exported by it load correctly in other viewers.

![Reference vs ObjViewer](https://user-images.githubusercontent.com/69405847/154867857-387a6574-e496-4b27-a0cc-1b94415a2f95.png)

I must say I don't know how the mapping is getting jumbled up.

### Reproduction

- Load any textured `obj` file in ObjectViewer.

### Related information 

* Operating system: Ubuntu 21.10 (Xubuntu) on x86_64
* Distribution: Windows/Linux archive
* Environment: Mono 6.8.0.105 (provided with `mono-complete`)

Additionally, a test model is attached.

[HCube.zip](https://github.com/leezer3/OpenBVE/files/8105098/HCube.zip)

**Comment by leezer3** [comment]
There are two texture co-ordinate systems for OBJ-
Left & right handed.

It's a lottery which one any given exporter / viewer supports (at least I don't think I've ever found a reliable way to detect)

We can probably hardcode Blockbench appropriately assuming it writes an identifiable header.
Will look at this later in the week :)

**Comment by Hibiii** [comment]
Ooohhhh that makes sense. Since OBJ supports comments, then perhaps a very quick and dirty solution could be something among the lines of `# openbve coordinates lefthand` or `righthand` on the first line, while there isn't a reliable way to detect it?

Have you tried tackling this problem before? Determining if an obj has a left-handed or right-handed coord system. What have you tried before? I want to help too, not just report, so anything would be nice. I gotta do my homework on WF obj though.

**Comment by leezer3** [comment]
https://github.com/leezer3/OpenBVE/issues/560

Later part of the discussion in this issue FWIW.
TLDR is that it's an utter mess :/

Latest commit to master will improve things somewhat, but the texture mappings (when using the Assimp parser option) are still a little broken, and I'm still investigating that one.

**Comment by leezer3** [comment]
https://vps.bvecornwall.co.uk/OpenBVE/Builds/OpenBVE-2022-02-23.zip

This build should now parse everything correctly I hope......
n.b. It uses the `# Made in Blockbench` header string to detect that it wants reversed handing, so this must not be removed.

If you've got any more complex models, it might be worth testing with these (not going to learn another 3D editor :P ), but I think things should now be OK.
